src/EventStore.Core/Services/ElectionsService.cs

```diff
@@ -71,6 +71,7 @@ public class ElectionsService : IHandle<SystemMessage.BecomeShuttingDown>,
 		private LeaderCandidate _leaderProposal;
 		private Guid? _leader;
 		private Guid? _lastElectedLeader;
+		public Guid TestElectedLeaderId = Guid.Empty; //this property is used for test visibility purposes only
```

Comment:
```suggestion
		public Guid ElectedLeaderId => _leader?.InstanceId ?? Guid.Empty;
```

It might be better to expose this properly rather than relying on setting a separate property just for testing.

Reply:
I've pushed a new commit to improve this part: cd3760cf4a3a6cad73949914ed562e0b119b7181

thanks!

Reply:
Yeah, I agree. There isn't much harm in exposing it properly as it is readonly.